doc/getting_started/index.rst

```diff
@@ -311,6 +311,42 @@ Set up Reference UOS
       :name: gsg-successful-boot
 
 
+Device Manager memory allocation mechanism
+==========================================
+
+There are two Device Manager memory allocation mechanisms available:
+
+- Contiguous Memory Allocator (CMA), and 
+- Huge Page Tables (HugeTLB).  HugeTLB is the default.
+
+To choose CMA, do the following:
+
+1) Add ``cma=reserved_mem_size@recommend_memory_offset-0``, (for example
+   ``cma=2560M@0x100000000-0``) to the SOS cmdline in ``acrn.conf``
+
+2) Start ``acrn-dm`` *without* the ``-T`` option
+
+To support HugeTLB, do the following:
+
+1) Do huge page reservation
+
+   - For 1G huge page reservation, add ``hugepagesz=1G hugepages=reserved_pg_num``
+     (for example, ``hugepagesz=1G hugepages=4``) to the SOS cmdline in
+     ``acrn.conf`` (for EFI)
+
+   - For 2M huge page reservation, after the SOS starts up, run the
+     command::
+
+        echo reserved_pg_num > /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages
+
+   .. note::
+      You can use 2M reserving method to do reservation for 1G page size, but it
+      may fail.  For an EFI platform, you may skip 1G page reservation
+      by using a 2M page, but make sure your huge page reservation size is
+      large enough for your usage.
```

Comment:
I do not have any prior knowledge on CMA vs. HugeTLB and I'm a little confused as to which I should use after reading this.

There is a large gap in terms of the amount of memory being reserved (2M vs. 1G). Can we add a little explanation (or a link to more info) on how to determine what to choose?

What would typically cause the 1G reservation (using the 2M method) to fail?

Would it make sense to use the 1G reservation method to reserve 2M?

Reply:
@gvancuts , hugetlb is new mechanism which should be preferred as CMA (old one) has exist problems like 
- fail to reserve
- share with other clients
but currently we still keep two solutions, so user can still choose.  As 1G page need big block of continuous memory, so if after system run for a while, it's difficult to find such big memory block if the physical ram size is not so big for the platform. Normally we are easy to reserve 2M page, so we can use normal way (echo pg# > nr_hugepages).

Reply:
Hi @JasonChenCJ , can you confirm if my understanding is correct? If so, I would like to add some text to our GSG myself. Thanks